stock_reserve_sale/view/sale.xml

```diff
@@ -0,0 +1,140 @@
+                >
+                    <field string="Reserved" name="reserves_count" widget="statinfo" />
+                    <field name="reserves_count" invisible="1" />
+                    <field name="all_lines_reserved" invisible="1" />
+                </button>
+                <button
+                    class="oe_stat_button"
+                    string="Stock Picking Reserve"
+                    type="object"
+                    name="action_view_reserves_stock_picking"
+                    icon="fa-truck"
+                    attrs="{'invisible': ['|', ('reserves_count', '=', 0),('state', '!=', 'draft')]}"
+                />
+            </xpath>
+            <button name="action_quotation_send" position="before">
+                <field name="is_stock_reservable" invisible="1" />
+                <button
+                    name="%(action_sale_stock_reserve)d"
+                    type="action"
+                    string="Reserve Stock"
+                    help="Pre-book products from stock"
+                    attrs="{'invisible': ['|', ('all_lines_reserved', '=', True), ('state', '!=', 'draft')]}"
+                />
+            </button>
+            <field name="order_line" position="attributes">
+                <attribute name="options">{"reload_on_button": 1}</attribute>
+            </field>
+            <xpath
+                expr="//field[@name='order_line']/form//field[@name='state']"
+                position="before"
+            >
+                <field name="reservation_ids" invisible="1" />
+                <button
+                    name="%(action_sale_stock_reserve)d"
+                    type="action"
+                    string="Reserve Stock"
+                    attrs="{'invisible': ['|', ('reservation_ids', '!=', []),
+                                                ('state', '!=', 'draft')]}"
+                />
+                <button
+                    name="release_stock_reservation"
+                    type="object"
+                    string="Release Reservation"
+                    attrs="{'invisible': ['|', ('reservation_ids', '=', []),
+                                                ('state', '!=', 'draft')]}"
+                />
+            </xpath>
+            <xpath
+                expr="//field[@name='order_line']/tree/field[@name='price_subtotal']"
+                position="after"
+            >
+                <field name="reservation_ids" invisible="1" />
+                <field name="is_stock_reservable" invisible="1" />
+                <button
+                    name="%(action_sale_stock_reserve)d"
+                    type="action"
+                    icon="fa-lock"
+                    attrs="{'invisible': [('is_stock_reservable', '=', False)]}"
+                />
+                <button
+                    name="release_stock_reservation"
+                    type="object"
+                    icon="fa-undo"
```

Comment:
`2023-08-14 21:09:09,612 1 WARNING mig odoo.addons.base.models.ir_ui_view: A button with icon attribute (fa-undo) must have title in its tag, parents, descendants or have text`